ksp_plugin/flight_plan.cpp

```diff
@@ -67,6 +67,8 @@ FlightPlan::FlightPlan(
 
   // Create a fork for the first coasting trajectory.
   segments_.emplace_back(root_->NewForkWithoutCopy(initial_time_));
+  coast_analysers_.push_back(make_not_null_unique<OrbitAnalyser>(
+      ephemeris_, DefaultHistoryParameters()));
```

Comment:
Instead of managing all these arrays in parallel, could we create a small `struct Segment` containing a trajectory and (optionally) an analyser?  That would be less error-prone and more future proof.

Reply:
No. We keep the analys*ers* when we recompute the segments (and the analys*es*).  As analysis is asynchronous, we want to keep the old analysis around until the new one is ready (this is especially critical when using the sliders: continuous changes to the flight plan are reflected in the UI with a slight delay, instead of the UI being constantly blank until the slider is released).
The analysers work more like the manœuvres in that respect: we insert and erase them when a manœuvre is added or removed, instead of pushing and popping them every time the flight plan changes as we do for the segments.